conversions/src/Data/Conversions.hs

```diff
@@ -183,7 +183,12 @@ instance (MonadError (BoundError Int Word32) m) => Conversion (m Word32) Int whe
   convert = checkedFromIntegralToBounded
 
 instance (MonadError (BoundError Int Word64) m) => Conversion (m Word64) Int where
-  convert = checkedFromIntegralToBounded
+  convert value = do
+    when (value < 0) $ throwError error
+
+    maybe (throwError error) pure $ checkedFromIntegral value
+      where
+        error = BoundError value
```

Comment:
comment: `boundError` binding is already in top level scope, I am therefore using `error`.